### PR TITLE
Update EOL versions (8.0-staging)

### PR DESCRIPTION
**Pull request by richlander**

### CODE REVIEW

**Review by carlossanlop — APPROVED** [review]
LGTM. @richlander If this is ready to merge, please do the honors and smash that squash button.

**Comment by richlander** [comment]
I have been waiting on a slow dev (me) to get new images created at build-prereqs. I have one last one left and then I can get back to these. I haven't forgotten about this and the other PR. I didn't realize when I started this that it would take so long. Good realization for next time.

**Comment by richlander** [comment]
Looks like libraries tests failing on Windows 10.

```bash
    System.Net.Http.WinHttpHandlerFunctional.Tests.BidirectionStreamingTest.BackwardsCompatibility_DowngradeToHttp11 [FAIL]
      Assert.Throws() Failure
      Expected: typeof(System.Exception)
      Actual:   (No exception was thrown)
      Stack Trace:
        /_/src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/BidirectionStreamingTest.cs(345,0): at System.Net.Http.WinHttpHandlerFunctional.Tests.BidirectionStreamingTest.BackwardsCompatibility_DowngradeToHttp11()
        --- End of stack trace from previous location ---
```

@wfurt

**Comment by danmoseley** [comment]
Do we need changes in any other repos?

**Comment by richlander** [comment]
Great q. Yes. I have been making changes across a variety of repos. Sadly, I keep on finding more images to remediate. It would be excellent if others want to help.

Examples: https://github.com/dotnet/aspnetcore/pulls?q=is%3Apr+author%3Arichlander+is%3Aclosed

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 1]
```diff
@@ -171,7 +171,7 @@ jobs:
 
     # WASI
     - ${{ if eq(parameters.platform, 'wasi_wasm') }}:
-      - (Ubuntu.2004.Amd64)Ubuntu.2204.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-20.04-helix-wasm-amd64
+      - (Ubuntu.1804.Amd64)Ubuntu.2204.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-wasm-amd64
```
Comment (akoeplinger) [inline eng/pipelines/libraries/helix-queues-setup.yml:174]: this doesn't look correct to me, it's downgrading the version

Comment (richlander) [reply eng/pipelines/libraries/helix-queues-setup.yml:174]: #105062

**Pull request merged**